staging/src/k8s.io/legacy-cloud-providers/azure/azure_client.go

```diff
@@ -738,6 +744,34 @@ func (az *azSecurityGroupsClient) CreateOrUpdate(ctx context.Context, resourceGr
 	return future.Response(), err
 }
 
+// createOrUpdatePreparer prepares the CreateOrUpdate request.
+func (az *azSecurityGroupsClient) createOrUpdatePreparer(ctx context.Context, resourceGroupName string, networkSecurityGroupName string, parameters network.SecurityGroup, etag string) (*http.Request, error) {
+	pathParameters := map[string]interface{}{
+		"networkSecurityGroupName": autorest.Encode("path", networkSecurityGroupName),
+		"resourceGroupName":        autorest.Encode("path", resourceGroupName),
+		"subscriptionId":           autorest.Encode("path", az.client.SubscriptionID),
+	}
+
+	const APIVersion = "2017-09-01"
```

Comment:
hardcoded? any reference?

Reply:
yep, the api version is hard coded. it is same from `"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"` now.

Reply:
did you copy this function `createOrUpdatePreparer` from go sdk? Looks like go sdk need to make this function as Public

Reply:
this is an internal function for NSG client, so it shouldn't be public.

There's indeed a preparer function in Go SDK, it's public because it may be invoked by SDK consumers.

Reply:
I know, why not use that public function in Go SDK? what's the diff between these two functions? You may paste the link of that function.

Reply:
createOrUpdatePreparer() added etag support, while Go SDK's one doesn't support it.